src/utils/clmfates_interfaceMod.F90

```diff
@@ -381,8 +379,20 @@ subroutine CLMFatesGlobals1(surf_numpft,surf_numcft,maxsoil_patches)
      ! and allocations on the FATES side, which require
      ! some allocations from CLM (like soil layering)
 
-     call SetFatesGlobalElements1(use_fates,surf_numpft,surf_numcft,var_reader)
 
+     !------------------------------------------------------------------------
+
+     ! It is FATES' job to open and close the file
+     ! This step is simply to provide a unit number to use
+     fates_paramfile_unit = shr_file_getUnit()
```

Comment:
shr_file_getUnit is actually deprecated now that Fortran has built-in file unit management. When you use open to open a file you can use newunit=<unit-name> to get a free unit. So it simplifies how this is done.

Reply:
oh that's nice, ok